src/commands/mute.ts

```diff
@@ -0,0 +1,112 @@
+
+import { ICommand, PermissionLevel } from "./base";
+import { Message, Client } from "discord.js";
+import ServerSettingsRepository from "../repository/serverSettings";
+import createMessageEmbed from "../wrapper/discord/messageEmbed";
+import MutedRepository from "../repository/muted";
+import { UnmuteWhenExpired } from '../lib/mutedRole';
+import { isReturnStatement } from "typescript";
+
+export default class MuteCommand implements ICommand {
+
+	commandName = 'mute';
+	aliases = null;
+	permissionLevel = PermissionLevel.Moderator;
+	guildOnly = false;
+
+	usageText = ";mute <user> <time> <reason>";
+	helpText = "Mutes user";
+	
+	async run(discordClient: Client, message: Message, args: string[]) {
+		const guildId = message.guild?.id;
+		const serverSettings = await ServerSettingsRepository.GetByGuildId(guildId);
+		if (serverSettings === null || serverSettings.muteRole === null) {
+			message.reply("Muting has not been configured");
+			return;
+		}
+
+		if (args.length < 3 || message.mentions.members === null || message.mentions.members.size === 0) {
+			message.reply(this.usageText);
+			return;
+		}
+
+		const guildMember = message.mentions.members.first()
+		if (!guildMember || !args[0].includes(guildMember.id)) {
+			message.reply(this.usageText)
+			return;
+		}
+		
+		if (guildMember.id === message.author.id) { 
+			message.reply("User can't mute themselves");
+			return;
+		}
+
+		const date = this.parseDate(args[1]);
```

Comment:
Muting for a longer time, like a month, results in a console error and the mute not working.

```
(node:926) TimeoutOverflowWarning: 2678400136 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
```